summaries/Evolutionary-testing-of-classes-summary.md

```diff
@@ -0,0 +1,81 @@
+# Evolutionary Testing of Classes
+**Article:**
+*Tonella, Paolo*. Evolutionary testing of classes. In Proceedings of the 2004 ACM SIGSOFT international symposium on Software testing and analysis (ISSTA '04). ACM, New York, NY, USA, 119-128.
+
+
+## Automated Test Case Generation
+Consists of creating an object, changing the state of that object and method call with proper input values. 
+Unit tests of classes should include all possible constructor and method invocation in order all cases to be analyzed.
+Generic algorithms use chromosome encoding ensuring that for hard to reach code fragments test cases are implemented by creating sequence of program calls.
+
+## Generic Algorithms for the Unit Testing of Classes
+Generic algorithms are broadly applied in testing procedural programs.
+
+### Unit testing of classes
+Unit testing of classes considers that each class being tested consists of constructor(s), method invocation within sertain set of input parameters
+
+### Evolutionary testing
+Evolutionary testing requires certain steps to be implemented in the algorithm including definition of parameters, maximum execution time, targets, automatically generated tests.
```

Comment:
`including definition` -> `including a definition`
`tagets, automatically` -> `targets and automatically`